test/src/kbasesearchengine/test/parse/ObjectParserTest.java

```diff
@@ -0,0 +1,256 @@
+package kbasesearchengine.test.parse;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.matchers.JUnitMatchers.both;
+import static org.junit.matchers.JUnitMatchers.containsString;
+import static org.junit.matchers.JUnitMatchers.hasItem;
+
+import java.io.File;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.commons.io.FileUtils;
+import org.ini4j.Ini;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.CharStreams;
+
+import kbasesearchengine.common.GUID;
+import kbasesearchengine.common.ObjectJsonPath;
+import kbasesearchengine.events.handler.SourceData;
+import kbasesearchengine.parse.IdMapper;
+import kbasesearchengine.parse.ObjectParser;
+import kbasesearchengine.parse.SimpleIdConsumer;
+import kbasesearchengine.parse.SimpleSubObjectConsumer;
+import kbasesearchengine.parse.SubObjectConsumer;
+import kbasesearchengine.system.ObjectTypeParsingRules;
+import us.kbase.common.service.UObject;
+
+public class ObjectParserTest {
+
+	private static Map<String, String> config = null;
+	private static Path scratch;
+	private static ObjectParser impl = null;
+
+	@BeforeClass
+	public static void init() throws Exception {
+		// config loading
+		String configFilePath = System.getenv("KB_DEPLOYMENT_CONFIG");
+		if (configFilePath == null) {
+			configFilePath = System.getProperty("KB_DEPLOYMENT_CONFIG");
+		}
+		File deploy = new File(configFilePath);
+		Ini ini = new Ini(deploy);
+		config = ini.get("KBaseSearchEngine");
+		scratch = Paths.get(config.get("scratch"), "test_object_parser");
+
+		impl = new ObjectParser();
+	}
+
+	@AfterClass
+	public static void teardown() throws Exception {
+		FileUtils.deleteDirectory(scratch.toFile());
+	}
+
+	@Test
+	public void prepareTempFileTest() throws Exception {
+
+		File tempDir = new File(scratch.toString(), UUID.randomUUID().toString());
+		assertEquals(false, tempDir.exists());
+
+		@SuppressWarnings("static-access")
+		File tmpFile = impl.prepareTempFile(tempDir);
+		assertEquals(true, tempDir.exists());
+		assertEquals(true, tmpFile.exists());
+		
+		assertThat(tmpFile.toString(), both(containsString("ws_srv_response_")).and(containsString(".json")));
+	}
+
+	@SuppressWarnings("static-access")
```

Comment:
What's causing this warning? I've never had to use it.

Reply:
👍